compiler/qsc_data_structures/src/telemetry.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+use std::sync::OnceLock;
+
+use serde::{Deserialize, Serialize};
+
+// See https://serde.rs/enum-representations.html
+// Over wasm, this will result in a JavaScript object similar to
+// { "id": "Parsed", "data": { "errors": 0, "size": 512 }}
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "id", content = "data")]
+pub enum Event<'a> {
+    Loaded { git_hash: &'a str },
+    Panic { details: &'a str },
+    Parsed { errors: u32, size: u64 },
+    Run { shots: u32, size: u64 },
+    Eval { qubits: u64, expression_count: u64 },
```

Comment:
Any intention to track any success/failure/perf metrics? Or are those for later PRs?

Just thinking through what queries we could write to gain insights from these events.

Reply:
Yeah, these are just some examples of what telemetry we _could_ emit to help structure the PR and its thinking with something concrete before they are actually added. Maybe just noise now though, or should be in comments if not used.

Reply:
Likewise, will check-in then worry about which telemetry to actually add after.